### PR TITLE
Bug 2115872: Fix unit-test failures in 4.10

### PR DESCRIPTION
**Pull request by abhat**
4.10 utest failure fixes.

### CODE REVIEW

File: go-controller/pkg/ovn/pods.go [hunk 1]
```diff
@@ -212,7 +212,12 @@ func (oc *Controller) deleteLogicalPort(pod *kapi.Pod, portInfo *lpInfo) (err er
 		// is not re-added into the cache. Delete logical switch port anyway.
 		annotation, err := util.UnmarshalPodAnnotation(pod.Annotations)
 		if err != nil {
-			return fmt.Errorf("unable to unmarshal pod annocations for pod %s/%s: %w", pod.Namespace, pod.Name, err)
+			if util.IsAnnotationNotSetError(err) {
```
Comment (abhat) [inline go-controller/pkg/ovn/pods.go:215]: This is pulling in the fix for https://bugzilla.redhat.com/show_bug.cgi?id=2092579. Apparently we had pulled in the utest that handles deleting logical port with no annotations in this commit: https://github.com/openshift/ovn-kubernetes/commit/b62dba23872c201b8542eaefa7bfa7697c0aaec1 but didn't bring the code that goes along. cc @anuragthehatter

**Comment by trozet** [comment]
@flavio-fernandes FYI this PR is pulling in a fix you did for 4.11 and later: https://github.com/openshift/ovn-kubernetes/commit/903660d2d103c984b382a65d2dc711479f9382b6

Looks like you accidentally pulled the unit test for this PR into another PR for 4.10: https://github.com/openshift/ovn-kubernetes/commit/b62dba23872c201b8542eaefa7bfa7697c0aaec1

@anuragthehatter @mffiedler be aware there is some minor functional change here, its not just unit test

File: go-controller/pkg/node/node_test.go [hunk 2]
```diff
@@ -366,7 +366,7 @@ var _ = Describe("Node", func() {
 						"external_ids:ovn-monitor-all=true "+
 						"external_ids:ovn-enable-lflow-cache=false "+
 						"external_ids:ovn-limit-lflow-cache=1000 "+
-						"external_ids:ovn-memlimit-lflow-cache-kb=100000",
+						"external_ids:ovn-limit-lflow-cache-kb=100000",
```
Comment (abhat) [inline go-controller/pkg/node/node_test.go:369]: @dcbw looks like the memlimit is the right parameter if i look at OVN docs, but then is our 4.10 code wrong? 
https://github.com/openshift/ovn-kubernetes/blob/release-4.10/go-controller/pkg/node/node.go#L221

Comment (abhat) [reply go-controller/pkg/node/node_test.go:369]: I wasn't sure if the field changed between the version of OVN we have in 4.10 vs. later. so just pushing this out. I can change the node setup too based on what is the right external-id for the flow cache size.

Comment (dcbw) [reply go-controller/pkg/node/node_test.go:369]: Supported external IDs in ovn21.12 (which 4.10 uses) are:

```
                           smap_get_bool(&cfg->external_ids,
                                         "ovn-enable-lflow-cache",
                                         true),
                           smap_get_uint(&cfg->external_ids,
                                         "ovn-limit-lflow-cache",
                                         DEFAULT_LFLOW_CACHE_MAX_ENTRIES),
                           smap_get_ullong(&cfg->external_ids,
                                           "ovn-memlimit-lflow-cache-kb",
                                           DEFAULT_LFLOW_CACHE_MAX_MEM_KB),
                           smap_get_uint(&cfg->external_ids,
                                         "ovn-trim-limit-lflow-cache",
                                         DEFAULT_LFLOW_CACHE_TRIM_LIMIT),
                           smap_get_uint(&cfg->external_ids,
                                         "ovn-trim-wmark-perc-lflow-cache",
                                         DEFAULT_LFLOW_CACHE_WMARK_PERC),
                           smap_get_uint(&cfg->external_ids,
                                         "ovn-trim-timeout-ms",
                                         DEFAULT_LFLOW_CACHE_TRIM_TO_MS));
```

So I guess we've got it wrong in ovnkube now?

Comment (dcbw) [reply go-controller/pkg/node/node_test.go:369]: Nope: we do it right in node.go, but the test is wrong:

```
go-controller/pkg/node/node.go:         fmt.Sprintf("external_ids:ovn-enable-lflow-cache=%t", config.Default.LFlowCacheEnable),
go-controller/pkg/node/node.go:                 fmt.Sprintf("external_ids:ovn-limit-lflow-cache=%d", config.Default.LFlowCacheLimit),
go-controller/pkg/node/node.go:                 fmt.Sprintf("external_ids:ovn-memlimit-lflow-cache-kb=%d", config.Default.LFlowCacheLimitKb),
```

Comment (dcbw) [reply go-controller/pkg/node/node_test.go:369]: @abhat I think this hunk to the test is wrong. Also upstream/master has it correct (eg, it has `ovn-memlimit-lflow-cache-kb` for that test, not what is in this PR)

Comment (abhat) [reply go-controller/pkg/node/node_test.go:369]: Only 4.10 has the problem.

Comment (abhat) [reply go-controller/pkg/node/node_test.go]: Reverted the utest fix and fixed the node code instead.

**Comment by flavio-fernandes** [comment]
I hit 1 test failure while executing `make test`. I will retry.

vagrant@devmaster ~/dev/ovn-kubernetes.git.ds/go-controller aniket_fix_utest4.10*
❯ make test

```
=== RUN   TestAdd/3:EgressFirewall_DNS_Run_Runs_update_after_the_ttl_returned_from_the_DNS_server_expires^M
    egressfirewall_dns_test.go:245: &{{0 0} 0xc0005737c0 map[] 0xc0003420a0 0xc0000486c0 0xc000c8cba0 0xc000048720 0xc000048660} <nil>^M
    egressfirewall_dns_test.go:277: timeout waiting for update based on ttl to fire or process^M
    egressfirewall_dns_test.go:285: &{{[0xc00049f5e0 0xc0000d7360 0xc000cc83c0 0xc000cc8500] [{0xc000342190 SetIPs [[2.2.2.2]] [] [AddressSet.go:168 egressfirewall_dns.go:131 egressfirewall_dns.go:145 asm_amd64.\
s:1581] 0 0 false <nil> 0 <nil> <nil>} {0xc000342190 SetIPs [[2.2.2.2 2001:db8:85a3::8a2e:370:7334]] [] [AddressSet.go:168 egressfirewall_dns.go:131 egressfirewall_dns.go:145 asm_amd64.s:1581] 0 0 false <nil> 0 \
<nil> <nil>} {0xc000342190 SetIPs [[2.2.2.2]] [] [AddressSet.go:168 egressfirewall_dns.go:131 egressfirewall_dns.go:145 asm_amd64.s:1581] 0 0 false <nil> 0 <nil> <nil>}] <nil> map[] {0 0}}} <nil>^M
    egressfirewall_dns_test.go:286: PASS:   ClientConfigFromFile(mock.AnythingOfTypeArgument)^M
    egressfirewall_dns_test.go:286: PASS:   ClientConfigFromFile(mock.AnythingOfTypeArgument)^M
    egressfirewall_dns_test.go:286: PASS:   Fqdn(mock.AnythingOfTypeArgument)^M
    egressfirewall_dns_test.go:286: PASS:   SetQuestion(mock.AnythingOfTypeArgument,mock.AnythingOfTypeArgument,mock.AnythingOfTypeArgument)^M
    egressfirewall_dns_test.go:286: PASS:   Exchange(mock.AnythingOfTypeArgument,mock.AnythingOfTypeArgument,mock.AnythingOfTypeArgument)^M
    egressfirewall_dns_test.go:286: PASS:   ClientConfigFromFile(mock.AnythingOfTypeArgument)^M
    egressfirewall_dns_test.go:286: PASS:   Fqdn(mock.AnythingOfTypeArgument)^M
    egressfirewall_dns_test.go:286: PASS:   Fqdn(mock.AnythingOfTypeArgument)^M
    egressfirewall_dns_test.go:286: PASS:   SetQuestion(mock.AnythingOfTypeArgument,mock.AnythingOfTypeArgument,mock.AnythingOfTypeArgument)^M
    egressfirewall_dns_test.go:286: PASS:   SetQuestion(mock.AnythingOfTypeArgument,mock.AnythingOfTypeArgument,mock.AnythingOfTypeArgument)^M
    egressfirewall_dns_test.go:286: PASS:   Exchange(mock.AnythingOfTypeArgument,mock.AnythingOfTypeArgument,mock.AnythingOfTypeArgument)^M
    egressfirewall_dns_test.go:286: PASS:   Exchange(mock.AnythingOfTypeArgument,mock.AnythingOfTypeArgument,mock.AnythingOfTypeArgument)^M
    egressfirewall_dns_test.go:286: PASS:   ClientConfigFromFile(mock.AnythingOfTypeArgument)^M
    egressfirewall_dns_test.go:286: PASS:   Fqdn(mock.AnythingOfTypeArgument)^M
    egressfirewall_dns_test.go:286: PASS:   SetQuestion(mock.AnythingOfTypeArgument,mock.AnythingOfTypeArgument,mock.AnythingOfTypeArgument)^M
    egressfirewall_dns_test.go:286: PASS:   Exchange(mock.AnythingOfTypeArgument,mock.AnythingOfTypeArgument,mock.AnythingOfTypeArgument)^M
    egressfirewall_dns_test.go:286: FAIL:   Fqdn(mock.AnythingOfTypeArgument)^M
                at: [egressfirewall_dns_test.go:214]^M
    egressfirewall_dns_test.go:286: FAIL:   SetQuestion(mock.AnythingOfTypeArgument,mock.AnythingOfTypeArgument,mock.AnythingOfTypeArgument)^M
                at: [egressfirewall_dns_test.go:214]^M
    egressfirewall_dns_test.go:286: FAIL:   Exchange(mock.AnythingOfTypeArgument,mock.AnythingOfTypeArgument,mock.AnythingOfTypeArgument)^M
                at: [egressfirewall_dns_test.go:214]^M
    egressfirewall_dns_test.go:286: FAIL: 16 out of 19 expectation(s) were met.^M
            The code you are testing needs to make 3 more call(s).^M
            at: [egressfirewall_dns_test.go:286]^M
    egressfirewall_dns_test.go:287: PASS:   NewAddressSet(mock.AnythingOfTypeArgument,mock.AnythingOfTypeArgument)^M
    egressfirewall_dns_test.go:287: PASS:   NewAddressSet(mock.AnythingOfTypeArgument,mock.AnythingOfTypeArgument)^M
    egressfirewall_dns_test.go:287: PASS:   NewAddressSet(mock.AnythingOfTypeArgument,mock.AnythingOfTypeArgument)^M
    egressfirewall_dns_test.go:287: PASS:   NewAddressSet(mock.AnythingOfTypeArgument,mock.AnythingOfTypeArgument)^M
    egressfirewall_dns_test.go:288: PASS:   SetIPs(mock.AnythingOfTypeArgument)^M
    egressfirewall_dns_test.go:288: PASS:   SetIPs(mock.AnythingOfTypeArgument)^M
    egressfirewall_dns_test.go:288: PASS:   SetIPs(mock.AnythingOfTypeArgument)^M
    egressfirewall_dns_test.go:288: FAIL:   SetIPs(mock.AnythingOfTypeArgument)^M
                at: [egressfirewall_dns_test.go:234]^M
    egressfirewall_dns_test.go:288: FAIL: 3 out of 4 expectation(s) were met.^M
            The code you are testing needs to make 1 more call(s).^M
            at: [egressfirewall_dns_test.go:288]^M
--- FAIL: TestAdd (15.03s)^M
    --- PASS: TestAdd/0:NewAddressSet_returns_error (0.00s)^M
    --- PASS: TestAdd/1:EgressFirewall_Add(dnsName)_succeeds_IPv4_only (0.01s)^M
    --- PASS: TestAdd/2:EgressFirewall_Add(dnsName)_succeeds_dual_stack (0.01s)^M
    --- FAIL: TestAdd/3:EgressFirewall_DNS_Run_Runs_update_after_the_ttl_returned_from_the_DNS_server_expires (15.01s)^M
```

Summarizing 1 Failure:

[Fail] OVN Pod Operations during execution [It] should not deallocate in-use and previously freed completed pods IP
/go/src/github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn/pods_test.go:560

Ran 153 of 153 Specs in 187.096 seconds
FAIL! -- 152 Passed | 1 Failed | 0 Pending | 0 Skipped
--- FAIL: TestClusterNode (187.33s)
FAIL
FAIL    github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn 210.787s
FAIL
make: *** [Makefile:51: test] Error 1

**Comment by flavio-fernandes** [comment]
but now I cannot reproduce the issue I hit on https://github.com/openshift/ovn-kubernetes/pull/1457#issuecomment-1353800259 ...  a classic flake. :)
TY for fixing the tests @abhat . This is awesome!
/lgtm

**Pull request merged**